### ISSUE TITLE
[Navigation Reboot] Create the ResponsiveNavigator

### ISSUE THREAD

**Issue by mountiny** [Expensify/App#15851]
Tracking issue https://github.com/Expensify/App/issues/11768

Created a new branch `navigation-refactor` to which we will merge all the PRs of this refactor.

Create the ResponsiveNavigator. In the first version we can keep the drawer to reduce PR size

Bookmark in [the doc](https://docs.google.com/document/d/1cQ87HcDA9RrmBM4ch7PmlNi9N9ZP8abQuPF0gKYrxDE/edit#bookmark=id.bpq4zejunk78)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01715c68be4674d12d</li>
        <li>Upwork Job ID: 1634918718682017792</li>
        <li>Last Price Increase: 2023-03-12</li>
    </ul>
</details>

**Comment by adamgrzybowski** [comment]
Going to implement this

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01715c68be4674d12d

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor Plus for review of internal employee PR - @mollfpr (`Internal`)

**Comment by adamgrzybowski** [comment]
Work will resume now once ECX is done

**Comment by mountiny** [comment]
This one will be done sooner as part of one PR with other two issues

**Comment by mountiny** [comment]
Unassigning CPlus for now, I will ask in the channel once the PR will be ready

**Comment by adamgrzybowski** [comment]
Still working on this. It's not that easy to create separate PR just for this issue so the Implementation will be pushed in a single PR including:

https://github.com/Expensify/App/issues/15848
https://github.com/Expensify/App/issues/15851
https://github.com/Expensify/App/issues/15852

**Comment by mountiny** [comment]
Actively being worked on

**Comment by MelvinBot** [comment]
@mountiny, @adamgrzybowski Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by MelvinBot** [comment]
@mountiny, @adamgrzybowski Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by mountiny** [comment]
Actively being worked on

**Comment by MelvinBot** [comment]
@mountiny, @adamgrzybowski Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by mountiny** [comment]
Actively being worked on

**Comment by mountiny** [comment]
Good to close
